### PR TITLE
Support ModuleSplitStyle in ScalaJSModule

### PR DESCRIPTION
**Pull request by lolgab**

### CODE REVIEW

File: scalajslib/worker/1/src/ScalaJSWorkerImpl.scala [hunk 1]
```diff
@@ -33,10 +34,11 @@ class ScalaJSWorkerImpl extends ScalaJSWorkerApi {
       fullOpt: Boolean,
       moduleKind: ModuleKind,
       esFeatures: ESFeatures,
+      moduleSplitStyle: ModuleSplitStyle,
       dest: File
   )
-  private def minorIsGreaterThanOrEqual(number: Int) = ScalaJSVersions.binaryEmitted match {
```
Comment (lolgab) [inline scalajslib/worker/1/src/ScalaJSWorkerImpl.scala:38]: `binaryEmitted` for Scala.js `1.10` is `1.8` for application level forward binary compatibility.
We want to know the build Scala.js version which can be accessed in current and looks like `1.10.0`

File: scalajslib/worker/1/src/ScalaJSWorkerImpl.scala [hunk 2]
```diff
@@ -94,12 +96,49 @@ class ScalaJSWorkerImpl extends ScalaJSWorkerApi {
...
+            )
+        }
+        config
+      }
+
+      config =
+        if (minorIsGreaterThanOrEqual(3)) withModuleSplitStyle_1_3_plus(config)
+        else withModuleSplitStyle_1_2_minus(config)
+
       StandardImpl.linker(config)
```
Comment (lefou) [inline scalajslib/worker/1/src/ScalaJSWorkerImpl.scala:142]: Could we use a new `val` here, to avoid the `var` above?

Comment (lolgab) [reply scalajslib/worker/1/src/ScalaJSWorkerImpl.scala]: Done ✅

File: scalajslib/src/mill/scalajslib/ScalaJSModule.scala [hunk 3]
```diff
@@ -83,11 +84,11 @@ trait ScalaJSModule extends scalalib.ScalaModule { outer =>
 
   def scalaJSToolsClasspath = T { scalaJSWorkerClasspath() ++ scalaJSLinkerClasspath() }
 
-  def fastLinkJS: Target[Report] = T {
+  def fastLinkJS: Target[Report] = T.persistent {
```
Comment (lolgab) [inline scalajslib/src/mill/scalajslib/ScalaJSModule.scala:87]: Asked about this on the Scala.js Discord:
Schermata 2022-05-05 alle 13 21 08

Comment (lefou) [reply scalajslib/src/mill/scalajslib/ScalaJSModule.scala:87]: (Hah, a nice example why discord isn't the best option for an open community. You can't easily link.)

Could you add a small comment about that behavior. This makes reasoning about the `T.persistent` much easier.

**Review by lefou — COMMENTED** [review]
Looks good to me, so far. But I'll wait for CI.

Thank you!

**Comment by lolgab** [comment]
Can we merge now ? :-)

**Review by lefou — APPROVED** [review]

**Pull request merged**